### PR TITLE
ts/indicators-macd

### LINKED ISSUE #11848: Migrate `indicators/macd` to TypeScript

### PR DESCRIPTION
**Pull request by activist**
Closes #11848, Tools: Migrated MACDIndicator to TypeScript.

### CODE REVIEW

**Comment by activist** [comment]
Further work awaits on the approval of #12124

File: ts/indicators/indicators.src.ts [hunk 1]
```diff
@@ -61,6 +61,12 @@ declare global {
...
 
+        interface IndicatorMultipleNullableValuesObject {
+            values: Array<Array<(number|null)>>;
+            xData: Array<(number|null)>;
+            yData: Array<Array<(number|null)>>;
+        }
+
         interface IndicatorMultipleValuesObject {
             values: IndicatorValuesObject['values'];
             xData: IndicatorValuesObject['xData'];
```
Comment (bre1470) [inline ts/indicators/indicators.src.ts]: For consistency you should inherit the type for `values` and `xData` in `IndicatorMultipleNullableValuesObject` the same way like in `IndicatorMultipleValuesObject` - or if you think it is wrong, instead replace the type inheritence in `IndicatorMultipleValuesObject`.

**Comment by bre1470** [comment]
BTW: The PR is still a draft.

File: ts/indicators/indicators.src.ts (hunk 1 above)
Comment (activist) [reply ts/indicators/indicators.src.ts]: Fixed as described first in your comment :)

**Comment by activist** [comment]
> Please resolve conflict

Fixed. The reason for the merge conflicts in this file is that Git considers also adjacent lines when calculating conflicts. So I reckon there unfortunately will be a few more of these.

**Pull request merged**